src/test/java/org/folio/rest/api/HoldingsSourceTest.java

```diff
@@ -0,0 +1,308 @@
+            .put("source", "marc")
+          );
+      UUID folioHoldingsSourceId = holdingsSource.getId();
+      UUID marcHoldingsSourceId = holdingsSource2.getId();
+
+      Response folioDeleteResponse = holdingsSourceClient.attemptToDelete(folioHoldingsSourceId);
+      Response marcDeleteResponse = holdingsSourceClient.attemptToDelete(marcHoldingsSourceId);
+
+      //they should not have been deleted:
+      assertThat(folioDeleteResponse.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+      assertThat(marcDeleteResponse.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+  }
+
+  @Test
+  public void canNotRemoveHoldingsSourcesAttachedToHoldings() throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+    UUID instanceId = UUID.randomUUID();
+    UUID sourceId = UUID.randomUUID();
+
+    JsonObject instanceToCreate = new JsonObject();
+
+    instanceToCreate.put("id", instanceId.toString());
+    instanceToCreate.put("source", "Test Source");
+    instanceToCreate.put("title", "Test Instance");
+    instanceToCreate.put("instanceTypeId", "535e3160-763a-42f9-b0c0-d8ed7df6e2a2");
+
+    instancesClient.create(instanceToCreate);
+
+    holdingsSourceClient.create(
+              new JsonObject()
+              .put("id", sourceId.toString())
+              .put("name", "associated source")
+            ).getJson();
+
+    holdingsClient.create(new HoldingRequestBuilder()
+      .withId(UUID.randomUUID())
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)
+      .withSource(sourceId));
+
+
+      Response sourceDeleteResponse = holdingsSourceClient.attemptToDelete(sourceId);
+
+      //the associated source should not have been deleted:
+      assertThat(sourceDeleteResponse.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+  }
+
+  @Test
+  public void canAssociateSourceWithHolding() throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+	    UUID instanceId = UUID.randomUUID();
+	    UUID sourceId = UUID.randomUUID();
+
+	    JsonObject instanceToCreate = new JsonObject();
+
+	    instanceToCreate.put("id", instanceId.toString());
+	    instanceToCreate.put("source", "Test Source");
+	    instanceToCreate.put("title", "Test Instance");
+	    instanceToCreate.put("instanceTypeId", "535e3160-763a-42f9-b0c0-d8ed7df6e2a2");
+
+	    instancesClient.create(instanceToCreate);
+
+	    holdingsSourceClient.create(
+	    	      new JsonObject()
+	    	      .put("id", sourceId.toString())
+	    	      .put("name", "associatable source")
+	    	    ).getJson();
+
+	    IndividualResource holdingsResponse = holdingsClient.create(new HoldingRequestBuilder()
+	      .withId(UUID.randomUUID())
+	      .forInstance(instanceId)
+	      .withPermanentLocation(mainLibraryLocationId)
+	      .withSource(sourceId));
+
+	    assertThat(holdingsResponse.getJson().getString("sourceId"), is(sourceId.toString()));
+  }
```

Comment:
Personally, I think this test could be part of the holdings tests rather than these tests